miden-tx/src/tests/kernel_tests/test_account.rs

```diff
@@ -587,19 +585,18 @@ fn test_get_vault_commitment() {
 // ================================================================================================
 
 #[test]
-fn test_authenticate_procedure() {
-    let tx_context = TransactionContextBuilder::with_standard_account(ONE).build();
-    let account = tx_context.tx_inputs().account();
+fn test_procedure_in_advice_provider() {
```

Comment:
I would keep the naming of this test `test_authenticate_procedure` because `authenticate_procedure` is a kernel proc. 

And it is the one that we are trying to test the logic of here.

Reply:
I changed it back, but there's a subtle distinction I think: Even though `authenticate_procedure` is the actually called procedure, it errors immediately where the `push_procedure` event is emitted instead of the check for comparing roots and authenticating against the account code.

```python
export.authenticate_procedure
    # load procedure index
    push.20897 drop                                     # TODO: remove line, see miden-vm/#1122
    emit.ACCOUNT_PUSH_PROCEDURE_INDEX_EVENT adv_push.1
    # => [index, PROC_ROOT]

    # get procedure info (PROC_ROOT, storage_offset) from memory stored at index
    exec.get_procedure_info
    # => [PROC_ROOT, storage_offset, PROC_ROOT]

    # verify that PROC_ROOT exists in memory at index
    movup.4 movdn.8 assert_eqw.err=ERR_PROC_NOT_PART_OF_ACCOUNT_CODE
    # => [storage_offset]
end
```

If I'm not wrong the account code could contain the procedure, but if the procedure is not in the advice provider, the execution would fail anyway (with the same error it throws now) and the test would pass anyway, which is why I thought the other name was a tiny bit more appropriate.